models/index.js

```diff
@@ -85,3 +85,20 @@ exports.updateVotesById = (instructions, article_id) => {
       return rows;
     });
   };
+
+  exports.deleteCommentById = (comment_id) => {
+    const queryStr = 
+      `DELETE FROM comments WHERE comment_id=$1;`
+    return db.query(queryStr,[comment_id]).then(({rows}) => {
+      return rows
+    });
+  };
+
+  exports.checkCommentExists = (comment_id) => {
+    const queryStr = `SELECT * FROM comments WHERE comment_id=$1;`;
+    return db.query(queryStr,[comment_id]).then(({ rows }) => {
+      if (rows.length === 0) {
+        return Promise.reject({ status: 404, msg: "not found" });
+      }
+    });
+  };
```

Comment:
Adding a RETURNING *; in your query would allow you to return all the rows and put this in one data query. This is not the most efficient way to do this. Using Promise.all does not guarantee the order in which its promises are resolved. Please look over this and refactor it accordingly.